docs/root/configuration/upstream/cluster_manager/cluster_stats.rst

```diff
@@ -245,6 +245,7 @@ are rooted at *cluster.<name>.* and contain the following statistics:
   upstream_rq_<\*xx>, Counter, "Aggregate HTTP response codes (e.g., 2xx, 3xx, etc.)"
   upstream_rq_<\*>, Counter, "Specific HTTP response codes (e.g., 201, 302, etc.)"
   upstream_rq_time, Histogram, Request time milliseconds
+  upstream_rq_first_upstream_data_ms, Histogram, "Time in milliseconds from request start to first DATA frame byte received from upstream (useful for time-to-first-token in streaming scenarios like gRPC)"
```

Comment:
There's an open issue (https://github.com/envoyproxy/envoy/issues/34168) discussing whether all the durations should be exported as metrics. One concern is that histograms are relatively expensive (memory use mainly). I think it's likely we'll want to put them all behind config. It may make sense to remove this histogram for now and wait for resolution on that issue.

Reply:
Yeah, it could either be an http filter, or just another config flag, but adding this one unconditionally would open the flood gates to the rest of them, or make them all inconsistent in config.

Reply:
Defered histogram to later. 
Can we have a config driven approach to constructing histograms from `COMMON_DURATION` s ?